app/models/user.rb

```diff
@@ -118,14 +118,15 @@ class User < ApplicationRecord
   scope :matches_email, ->(value) { where(arel_table[:email].matches("#{value}%")) }
   scope :matches_ip, ->(value) { left_joins(:ips).where('user_ips.ip <<= ?', value).group('users.id') }
 
-  before_validation :sanitize_languages
   before_validation :sanitize_role
-  before_validation :sanitize_time_zone
-  before_validation :sanitize_locale
   before_create :set_approved
   after_commit :send_pending_devise_notifications
   after_create_commit :trigger_webhooks
 
+  normalizes :locale, with: ->(locale) { (I18n.available_locales.map(&:to_s) & [locale]).first }
+  normalizes :time_zone, with: ->(time_zone) { (ActiveSupport::TimeZone.all.map(&:name) & [time_zone]).first }
+  normalizes :chosen_languages, with: ->(chosen_languages) { chosen_languages.compact_blank!.presence }
```

Comment:
Sorry, but I think the ternary forms for the first two were both more efficient and easier to read :sweat_smile:

I'd suggest:
```suggestion
  normalizes :locale, with: ->(locale) { I18n.available_locales.exclude?(locale.to_sym) ? nil : locale }
  normalizes :time_zone, with: ->(time_zone) { ActiveSupport::TimeZone[time_zone].nil? ? nil : time_zone }
  normalizes :chosen_languages, with: ->(chosen_languages) { chosen_languages.compact_blank.presence }
```